aws/cf_template.go

```diff
@@ -478,6 +478,11 @@ func newTargetGroup(spec *stackSpec, targetPortParameter string) *cloudformation
 	// custom target group healthcheck only supported when the target group protocol is != TCP
 	if protocol != "TCP" {
 		targetGroup.HealthCheckTimeoutSeconds = cloudformation.Ref(parameterTargetGroupHealthCheckTimeoutParameter).Integer()
+		if spec.http2 {
+			targetGroup.ProtocolVersion = cloudformation.String("HTTP2")
+		} else {
+			targetGroup.ProtocolVersion = cloudformation.String("HTTP1")
+		}
```

Comment:
`spec.http2` is true by default. If I get it right this change would enable HTTP/2 on the target group and according to this table https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#target-group-protocol-version



Request protocol | Protocol version | Result
-- | -- | --
HTTP/1.1 | HTTP/2 | Error
HTTP/2 | HTTP/2 | Success

this change
* is not backward-compatible as default protocol version is HTTP/1.1
* would not work if target supports only HTTP/1.1 
* will disable HTTP/1.1 incoming requests

If that is correct then we may probably need to have a new flag and/or annotation to steer Target Group protocol version independently and backward-compatible.

Besides that HTTP/2 is only supported in conjunction with `HTTPS` protocol so this kind of config should be around here
https://github.com/zalando-incubator/kube-ingress-aws-controller/blob/5c661370ed2aa4cf2a0f4164f5a9114c80fe2c84/aws/cf_template.go#L455-L458

Reply:
This is true, the http/2 TG protocol is a breaking change for http/1.1 clients. Looking for a procedure how to deal with while supporting end-to-end, found this statement: https://stackoverflow.com/a/66800142

Probably going to open a case at AWS

Reply:
well known issue: https://forums.aws.amazon.com/message.jspa?messageID=985494

Reply:
It also shows the complexity that everything is based on the client. I am thinking for a while to support h2c in skipper, but the current implementations have some critical bugs and if we would be able to serve it then I also do not know if grpc backends can listen on handlers without TLS....

Reply:
@AlexanderYastrebov interestingly the protocol H2 can be set via AWS console also for non-https target groups, also I found that documented in the TF provider https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_target_group#protocol_version

Nevertheless I implemented as requested